server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -1001,10 +1001,8 @@ func (e *EventChannel) handleEvent(event *pepb.PublishBuildToolEventStreamReques
 	// Bazel sends an Interrupted exit code in the finished event if the user cancelled the build.
 	// Use that signal to cancel any actions that are currently in the remote execution system.
 	if f, ok := bazelBuildEvent.GetPayload().(*build_event_stream.BuildEvent_Finished); ok {
-		if f.Finished.GetExitCode().GetCode() == InterruptedExitCode && e.env.GetRemoteExecutionService() != nil {
-			if err := e.env.GetRemoteExecutionService().Cancel(e.ctx, iid); err != nil {
-				log.CtxWarningf(e.ctx, "Could not cancel executions for invocation %q: %s", iid, err)
-			}
+		if err := e.env.GetRemoteExecutionService().Cancel(e.ctx, iid); err != nil {
```

Comment:
@vadimberezniker Is this safe to do with action merging?

Reply:
I don't think so. I think even the interrupt cancellation we do today does not handle merging properly.

Reply:
OK, so need to resolve that first.